proto/internal/uber/cadence/indexer/v1/messages.proto

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package uber.cadence.indexer.v1;
+
+option go_package = "indexerv1";
+
+message Message {
+  MessageType message_type = 1;
+  string namespace_id = 2;
+  string workflow_id = 3;
+  string run_id = 4;
+  int64 version = 5;
+  map<string, Field> fields = 6;
+}
+
+message Field {
+  FieldType type = 1;
+  oneof data {
+    string string_data = 2;
+    int64 int_data = 3;
+    bool bool_data = 4;
+    bytes binary_data = 5;
```

Comment:
irrelevant for this set of changes, but:
a canonical timestamp field would probably be a good idea, so users don't have to come up with a string format or decide on what precision to use for integers.